packages/components/src/calendar/test/index.tsx

```diff
@@ -0,0 +1,751 @@
+			const user = setupUserEvent();
+
+			render( <DateCalendar disableNavigation /> );
+
+			expect( screen.getByRole( 'button', { name: /previous month/i } ) ).toHaveAttribute(
+				'aria-disabled',
+				'true'
+			);
+			expect( screen.getByRole( 'button', { name: /next month/i } ) ).toHaveAttribute(
+				'aria-disabled',
+				'true'
+			);
+
+			await user.tab();
+			expect( screen.getByRole( 'button', { name: /today/i } ) ).toHaveFocus();
+		} );
+	} );
+
+	// Note: we're not testing localization of strings. We're only testing
+	// that the date formatting, computed dir, and calendar format are correct.
+	describe( 'Localization', () => {
+		it( 'should localize the calendar based on the `locale` prop', async () => {
+			const user = setupUserEvent();
+
+			render( <DateCalendar locale={ ar } /> );
+
+			// Check computed writing direction
+			expect( screen.getByRole( 'application', { name: 'Date calendar' } ) ).toHaveAttribute(
+				'dir',
+				'rtl'
+			);
+
+			// Check month name
+			const grid = screen.getByRole( 'grid', {
+				name: monthNameFormatter( 'ar' ).format( today ),
+			} );
+			expect( grid ).toBeInTheDocument();
+
+			// Check today button
+			expect( getDateButton( today, {}, 'ar' ) ).toHaveAccessibleName( /today/i );
+
+			await user.tab();
+			await user.tab();
+			await user.tab();
+			expect( getDateButton( today, {}, 'ar' ) ).toHaveFocus();
+
+			await user.keyboard( '{Home}' );
+			expect( getDateButton( startOfWeek( today, { locale: ar } ), {}, 'ar' ) ).toHaveFocus();
+		} );
+
+		it( 'should support timezones according to the `timeZone` prop', async () => {
+			const user = setupUserEvent();
+			const onSelect = jest.fn();
+
+			render( <DateCalendar timeZone="Asia/Tokyo" onSelect={ onSelect } /> );
+
+			// For someone in Tokyo, 20:00 UTC is the next day.
+			expect( getDateButton( tomorrow ) ).toHaveAccessibleName( /today/i );
+
+			// Select tomorrow's button (which is today in Tokyo)
+			const tomorrowButton = getDateButton( tomorrow );
+			await user.click( tomorrowButton );
+			expect( onSelect ).toHaveBeenCalledWith(
+				fromZonedTime( tomorrow, 'Asia/Tokyo' ),
+				fromZonedTime( tomorrow, 'Asia/Tokyo' ),
+				expect.objectContaining( { today: true } ),
+				expect.objectContaining( { type: 'click', target: tomorrowButton } )
+			);
+		} );
+
+		it( 'should handle timezoned dates and convert them to the calendar timezone', async () => {
+			// Still the same time from UTC's POV, just expressed in Tokyo time.
+			const tomorrowAtMidnightInTokyo = new TZDate( tomorrow, 'Asia/Tokyo' );
+
+			render( <DateCalendar defaultSelected={ tomorrowAtMidnightInTokyo } timeZone="-02:00" /> );
+
+			// Changing the calendar timezone to UTC-2 makes the dates become
+			// earlier by 1 day (from midnight to 10pm the previous day).
+			expect( getDateCell( today, { selected: true } ) ).toBeVisible();
+		} );
+	} );
```

Comment:
I wonder how much edge cases we want to support. Like, should we handle daylight savings time transitions correctly?

Reply:
Sounds good to me, thanks for confirming!